To: vim_dev@googlegroups.com
Subject: Patch 7.3.293
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.293
Problem:    MSVC compiler has a problem with non-ASCII characters.
Solution:   Avoid non-ASCII characters. (Hong Xu)
Files:	    src/ascii.h, src/spell.c


*** ../vim-7.3.292/src/ascii.h	2011-05-10 16:41:13.000000000 +0200
--- src/ascii.h	2011-09-02 14:12:48.000000000 +0200
***************
*** 123,129 ****
  #define DCS		0x90	/* Device Control String */
  #define STERM		0x9c	/* String Terminator */
  
! #define POUND		'£'
  
  #define CTRL_F_STR	"\056"
  #define CTRL_H_STR	"\026"
--- 123,129 ----
  #define DCS		0x90	/* Device Control String */
  #define STERM		0x9c	/* String Terminator */
  
! #define POUND		'\xA3'
  
  #define CTRL_F_STR	"\056"
  #define CTRL_H_STR	"\026"
*** ../vim-7.3.292/src/spell.c	2011-04-11 21:35:03.000000000 +0200
--- src/spell.c	2011-09-02 14:13:30.000000000 +0200
***************
*** 11197,11203 ****
  	    c = *s++;
  
  #ifdef FEAT_MBYTE
! 	/* We only change ß to SS when we are certain latin1 is used.  It
  	 * would cause weird errors in other 8-bit encodings. */
  	if (enc_latin1like && c == 0xdf)
  	{
--- 11197,11203 ----
  	    c = *s++;
  
  #ifdef FEAT_MBYTE
! 	/* We only change 0xdf to SS when we are certain latin1 is used.  It
  	 * would cause weird errors in other 8-bit encodings. */
  	if (enc_latin1like && c == 0xdf)
  	{
*** ../vim-7.3.292/src/version.c	2011-09-02 14:07:31.000000000 +0200
--- src/version.c	2011-09-02 14:13:57.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     293,
  /**/

-- 
   A village.  Sound of chanting of Latin canon, punctuated by short, sharp
   cracks.  It comes nearer.  We see it is a line of MONKS ala SEVENTH SEAL
   flagellation scene, chanting and banging themselves on the foreheads with
   wooden boards.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
